### PR TITLE
Add updateApprovalStatus in ApplicantRepository

### PR DESCRIPTION
**Pull request by BlancoSebastianEzequiel**

### CODE REVIEW

File: test/models/Applicant/Repository.test.ts [hunk 1]
```diff
@@ -583,4 +586,190 @@ describe("ApplicantRepository", () => {
...
+      ]);
+    };
+
+    it("creates an applicant with a pending status", async () => {
+      const applicant = await ApplicantRepository.create(applicantsMinimumData.next().value);
+      expect(applicant.approvalStatus).toEqual(ApprovalStatus.pending);
+    });
+
+    it("approves applicant only by an admin", async () => {
+      await expectApplicantWithApprovalStatus(ApprovalStatus.approved);
```
Comment (dylanalvarez) [inline test/models/Applicant/Repository.test.ts:628]: expectSucessfulApplicantStatusUpdate

File: test/models/Applicant/Repository.test.ts [hunk 2]
```diff
@@ -583,4 +586,190 @@ describe("ApplicantRepository", () => {
...
+        })
+      ]);
+    };
+
+    it("creates an applicant with a pending status", async () => {
+      const applicant = await ApplicantRepository.create(applicantsMinimumData.next().value);
+      expect(applicant.approvalStatus).toEqual(ApprovalStatus.pending);
+    });
+
+    it("approves applicant only by an admin", async () => {
```
Comment (dylanalvarez) [inline test/models/Applicant/Repository.test.ts:627]: "allows admins to approve an applicant"

File: test/models/Applicant/Repository.test.ts [hunk 3]
```diff
@@ -583,4 +586,190 @@ describe("ApplicantRepository", () => {
...
+      );
+    };
+
+    const expectApplicantWithApprovalStatus = async (approvalStatus: ApprovalStatus) => {
+      const applicant = await ApplicantRepository.create(applicantsMinimumData.next().value);
+      const approvedApplicant = await updateApplicantWithStatus(applicant, approvalStatus);
+      expect(approvedApplicant.approvalStatus).toEqual(approvalStatus);
+    };
+
+    const expectApplicantToHaveOneApprovalEvent = async (approvalStatus: ApprovalStatus) => {
```
Comment (dylanalvarez) [inline test/models/Applicant/Repository.test.ts:610]: expectApplicantApprovalEventCreation ?

Comment (dylanalvarez) [reply test/models/Applicant/Repository.test.ts:610]: expectStatusUpdateToCreateOneEvent

File: test/models/Applicant/Repository.test.ts [hunk 4]
```diff
@@ -583,4 +586,190 @@ describe("ApplicantRepository", () => {
...
+    it("creates an applicant with a pending status", async () => {
+      const applicant = await ApplicantRepository.create(applicantsMinimumData.next().value);
+      expect(applicant.approvalStatus).toEqual(ApprovalStatus.pending);
+    });
+
+    it("approves applicant only by an admin", async () => {
+      await expectApplicantWithApprovalStatus(ApprovalStatus.approved);
+    });
+
+    it("rejects applicant only by an admin", async () => {
```
Comment (dylanalvarez) [inline test/models/Applicant/Repository.test.ts:631]: allows admins to reject an applicant

File: test/models/Applicant/Repository.test.ts [hunk 5]
```diff
@@ -583,4 +586,190 @@ describe("ApplicantRepository", () => {
...
+      await expectApplicantWithApprovalStatus(ApprovalStatus.rejected);
+    });
+
+    it("returns approved event by association", async () => {
```
Comment (dylanalvarez) [inline test/models/Applicant/Repository.test.ts:635]: creates an event when approving an applicant

File: test/models/Applicant/Repository.test.ts [hunk 6]
```diff
@@ -583,4 +586,190 @@ describe("ApplicantRepository", () => {
...
+      await expectApplicantToHaveOneApprovalEvent(ApprovalStatus.approved);
+    });
+
+    it("returns pending event by association", async () => {
```
Comment (dylanalvarez) [inline test/models/Applicant/Repository.test.ts:639]: creates an event when marking an applicant approval as pending

File: test/models/Applicant/Repository.test.ts [hunk 7]
```diff
@@ -583,4 +586,190 @@ describe("ApplicantRepository", () => {
...
+      await expectApplicantToHaveOneApprovalEvent(ApprovalStatus.pending);
+    });
+
+    it("returns rejected event by association", async () => {
```
Comment (dylanalvarez) [inline test/models/Applicant/Repository.test.ts:643]: creates an event when rejecting an applicant

File: test/models/Applicant/Repository.test.ts [hunk 8]
```diff
@@ -583,4 +586,190 @@ describe("ApplicantRepository", () => {
...
+          applicant.uuid,
+          "notDefinedValue" as ApprovalStatus
+        )
+      ).rejects.toThrowErrorWithMessage(
+        ValidationError,
+        `ApprovalStatus must be one of these values: ${approvalStatuses}`
+      );
+    });
+
+    it("doest not update applicant status if it throws an error", async () => {
```
Comment (dylanalvarez) [inline test/models/Applicant/Repository.test.ts:749]: does not

Comment (dylanalvarez) [reply test/models/Applicant/Repository.test.ts:749]: does not change anything if an error is thrown

File: test/models/Applicant/Repository.test.ts [hunk 9]
```diff
@@ -583,4 +586,190 @@ describe("ApplicantRepository", () => {
...
+          applicant.uuid,
+          "unknownStatus" as ApprovalStatus
+        )
+      ).rejects.toThrow();
+      expect(
+        (await ApplicantRepository.findByUuid(applicant.uuid)).approvalStatus
+      ).toEqual(ApprovalStatus.pending);
+    });
+
+    it("doest not create an event for the applicant if it throws an error", async () => {
```
Comment (dylanalvarez) [inline test/models/Applicant/Repository.test.ts:763]: does not

File: src/models/Applicant/Repository.ts [hunk 10]
```diff
@@ -84,5 +86,30 @@ export const ApplicantRepository = {
       throw error;
     }
   },
+  updateApprovalStatus: async (
```
Comment (llmanuel) [inline src/models/Applicant/Repository.ts:89]: Para mantener la consistencia con lo que ya hicimos, aprovecharia a tener este metodo unicamente en un `ApplicantApprovalEventRepository`

**Pull request merged**